gadopt/level_set_tools.py

```diff
@@ -77,35 +77,48 @@ def assign_level_set_values(
     epsilon: float | fd.Function,
     /,
     interface_geometry: str,
-    interface_coordinates: list[list[float]] | list[list[float], float] | None = None,
     *,
+    interface: sl.LineString | sl.Polygon | None = None,
+    interface_coordinates: list[list[float]] | list[list[float], float] | None = None,
     interface_callable: Callable | str | None = None,
     interface_args: tuple[Any] | None = None,
```

Comment:
```suggestion
    interface_args: tuple[Any, ...] | None = None,
```
While we're on pedantic type checking, `interface_args` has unknown length, so it should be annotated with `...`.